var/spack/repos/builtin/packages/scorep/package.py

```diff
@@ -34,18 +34,20 @@ class Scorep(AutotoolsPackage):
     homepage = "http://www.vi-hps.org/projects/score-p"
     url      = "http://www.vi-hps.org/upload/packages/scorep/scorep-2.0.2.tar.gz"
 
+    version('4.1',   '3acad79cedd5c8c1456b0a8521760ba5')
     version('3.1',   '065bf8eb08398e8146c895718ddb9145')
     version('3.0',   '44da8beaa3f71436a5f6fe51938aab2f')
     version('2.0.2', '8f00e79e1b5b96e511c5ebecd10b2888')
     version('1.4.2', '3b9a042b13bdd5836452354e6567f71e')
     version('1.3',   '9db6f957b7f51fa01377a9537867a55c')
 
-    patch('gcc7.patch')
+    patch('gcc7.patch',when='gcc%7i:')
```

Comment:
Not sure about the syntax here for the `when=`, did you mean this?
```suggestion
    patch('gcc7.patch', when='%gcc@7:')
```
(A space was missing a space which makes flake8 unhappy.)